tasks/lib/postcss-notnested.js

```diff
@@ -0,0 +1,25 @@
+var postcss = require('postcss');
+
+module.exports = postcss.plugin('postcss-notnested', function (opts) {
+  opts = opts || {};
+
+  // Match ampersands at the start of a given selector
+  var re = /^&.*/m;
+
+  return function (root, result) {
+    root.walkRules((rule, ruleIndex) => {
+      var matchFound = false;
+      if (rule.selectors) {
+        var selectors = rule.selectors.filter(selector => {
+          // Kill the selector with the stray ampersand -- it's not nested!
+          return !re.test(selector)
+        });
+
+        // Only replace the selectors if we changed something (avoids extra work for every selector)
+        if (selectors.length != rule.selectors.length) {
```

Comment:
This avoids running this code for every single selector in the project: https://github.com/postcss/postcss/blob/8157278c743564555a22da67da2b5972945462d8/lib/rule.es6#L42-L46

Reply:
It also seems like, if length goes to zero, we should remove the rule completely... That way, we can add a single selector that assumes nesting, and the rule is killed if its not nested...